### PR TITLE
PROD-76: Add news model

### PR DESCRIPTION
**Pull request by Retttro**
# Reviewers

- [x] @vikabelous
- [x] @okohuch
- [x] @shevchukm
- [x] @VitaliiChverenko

# Jira task
http://ssu-jira.softserveinc.com/browse/PROD-76

# Description
Create news model, make migration
Create validations
Create tests for model
Add news seeds
image

# Checklist:

- [x] Rubocop doesn't raise any warnings and offenses
- [x] All specs pass
- [x] New features are covered with new specs
- [x] PR is approved by 2 team mates
- [x] PR is approved by 2 experts

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -20,4 +20,4 @@ Metrics/MethodLength:
   Exclude:
     - 'db/migrate/*'
 Style/FrozenStringLiteralComment:
-  Enabled: false
+  Enabled: false
```
Comment (shevchukm) [inline .rubocop.yml]: add empty line at the end

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -11,4 +11,4 @@
   <body>
     <%= yield %>
   </body>
-</html>
+</html>
```
Comment (shevchukm) [inline app/views/layouts/application.html.erb]: here too

File: config/database.yml.example [hunk 3]
```diff
@@ -1,23 +0,0 @@
...
-  <<: *default
-  database: ProDobro_test
-
-production:
-  adapter: postgresql
-  encoding: utf8
-  database: ProDobro_production
-  pool: 4
-  username:
-  password:
```
Comment (shevchukm) [inline config/database.yml.example]: why have you deleted it? return it back

File: spec/models/news_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  context 'When description field is empty' do
+    it 'is invalid' do
+      news.description = ''
+      expect(news).to_not be_valid
+    end
+  end
+
+  context 'When title field is to short' do
```
Comment (shevchukm) [inline spec/models/news_spec.rb]: maybe too short?

**Review by shevchukm — APPROVED** [review]

**Review by VitaliiChverenko — APPROVED** [review]

File: spec/factories.rb [hunk 5]
```diff
@@ -6,4 +6,8 @@
     email Faker::Internet.email
     password '369852'
   end
+  factory :news do
```
Comment (okohuch) [inline spec/factories.rb]: Each model should have its own factory file.

File: db/seeds.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+News.create(title: 'New title 1', description: Faker::Lorem.paragraph(4))
+
+News.create(title: 'New title 2', description: Faker::Lorem.paragraph(5))
```
Comment (okohuch) [inline db/seeds.rb]: Why do you need these seeds?

File: spec/factories.rb (hunk 5 above)
Comment (okohuch) [reply spec/factories.rb]: split this file into two

File: db/seeds.rb (hunk 6 above)
Comment (Retttro) [reply db/seeds.rb]: To add ability for others populate db faster

File: spec/factories.rb (hunk 5 above)
Comment (Retttro) [reply spec/factories.rb]: one file for users  and another for news?

Comment (okohuch) [reply spec/factories.rb]: yes

File: config/database.yml.example [hunk 7]
```diff
@@ -2,7 +2,11 @@ default: &default
   adapter: mysql2
   encoding: utf8
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+<<<<<<< HEAD
```
Comment (okohuch) [inline config/database.yml.example]: what is it??

File: db/schema.rb [hunk 8]
```diff
@@ -34,6 +34,61 @@
...
+
+ActiveRecord::Schema.define(version: 20180405090900) do
+  create_table "news", force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8" do |t|
+    t.string "title"
+    t.text "description"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
   end
 
+  create_table "users", force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8" do |t|
```
Comment (okohuch) [inline db/schema.rb]: two user tables?

File: db/schema.rb [hunk 9]
```diff
@@ -34,6 +34,61 @@
...
+    t.datetime 'confirmed_at'
+    t.datetime 'confirmation_sent_at'
+    t.string 'unconfirmed_email'
+    t.datetime 'created_at', null: false
+    t.datetime 'updated_at', null: false
+    t.index ['confirmation_token'], name: 'index_users_on_confirmation_token', unique: true
+    t.index ['email'], name: 'index_users_on_email', unique: true
+    t.index ['reset_password_token'], name: 'index_users_on_reset_password_token', unique: true
+
+ActiveRecord::Schema.define(version: 20180405090900) do
```
Comment (okohuch) [inline db/schema.rb]: two schemas?

File: config/database.yml.example (hunk 7 above)
Comment (Retttro) [reply config/database.yml.example]: its in fixing process)

**Comment by okohuch** [comment]
@Retttro read about how to do rebasing and conflict fixing

File: config/database.yml.example (hunk 7 above)
Comment (okohuch) [reply config/database.yml.example]: always check it locally via git diff before pushing it

**Comment by okohuch** [comment]
@Retttro I am not one of the approvers. Ping @vikabelous

File: spec/models/news_spec.rb [hunk 10]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe(News, type: :model) do
+  let(:news) { build(:news) }
+
+  context 'When title field is empty' do
```
Comment (omaik) [inline spec/models/news_spec.rb]: group together title and description specs.

File: db/seeds.rb (hunk 6 above)
Comment (viktoriia-bilous) [reply db/seeds.rb]: Use loop if you need to create more than one instance

File: spec/models/news_spec.rb [hunk 11]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe(News, type: :model) do
+  let(:news) { build(:news) }
+
+  context 'When title and description fields are empty' do
```
Comment (viktoriia-bilous) [inline spec/models/news_spec.rb]: lowercase 'When'

File: spec/models/news_spec.rb [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    it 'is invalid' do
+      news.title = ''
+      news.description = ''
+      expect(news).to_not be_valid
+      expect(news).to_not be_valid
```
Comment (viktoriia-bilous) [inline spec/models/news_spec.rb]: one expectation per test case, split it

File: app/models/news.rb [hunk 13]
```diff
@@ -0,0 +1,4 @@
+class News < ApplicationRecord
+  validates :title, presence: true, length: { minimum: 3 }
```
Comment (okohuch) [inline app/models/news.rb]: add relation with user model

Comment (Retttro) [reply app/models/news.rb]: @okohuch added, now trying to create specs

File: db/migrate/20180415221221_add_user_id_to_news.rb [hunk 14]
```diff
@@ -0,0 +1,5 @@
+class AddUserIdToNews < ActiveRecord::Migration[5.1]
+  def change
+    add_column :news, :user_id, :integer
```
Comment (okohuch) [inline db/migrate/20180415221221_add_user_id_to_news.rb]: no need to have separate migration
add this column while creating news table

File: db/schema.rb [hunk 15]
```diff
@@ -10,7 +10,17 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20180404162953) do
+
```
Comment (okohuch) [inline db/schema.rb]: why so many empty lines?

File: spec/rails_helper.rb [hunk 16]
```diff
@@ -20,3 +20,9 @@
 
   config.filter_rails_from_backtrace!
 end
+Shoulda::Matchers.configure do |config|
```
Comment (okohuch) [inline spec/rails_helper.rb]: add an empty line between configurations

File: spec/models/news_spec.rb [hunk 17]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe(News, type: :model) do
+  let(:news) { create(:news) }
+
+  context 'when title and description fields are empty' do
```
Comment (okohuch) [inline spec/models/news_spec.rb]: you have added https://github.com/thoughtbot/shoulda-matchers so please rewrite all tests as we agreed

**Review by okohuch — CHANGES_REQUESTED** [review]
address the comments

File: db/migrate/20180405090900_create_news.rb [hunk 18]
```diff
@@ -0,0 +1,11 @@
+class CreateNews < ActiveRecord::Migration[5.1]
+  def change
+    create_table :news do |t|
+      t.string :title
+      t.text :description
+      t.integer :news, :user_id
```
Comment (okohuch) [inline db/migrate/20180405090900_create_news.rb]: did you run the migration?

File: spec/models/news_spec.rb [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+  context 'when title field is empty or too short' do
+    it { should_not allow_values('', 'No').for(:title) }
+  end
+
+  context 'when description field is empty' do
+    it { should_not allow_value('').for(:description) }
+  end
+
+  context 'when title field is minimum length' do
+    it { should allow_value('Yes').for(:title) }
```
Comment (okohuch) [inline spec/models/news_spec.rb]: https://github.com/thoughtbot/shoulda-matchers/blob/master/lib/shoulda/matchers/active_model/validate_length_of_matcher.rb#L149

File: spec/models/news_spec.rb [hunk 20]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe(News, type: :model) do
+  let(:news) { create(:news) }
+
+  context 'when title field is empty or too short' do
+    it { should_not allow_values('', 'No').for(:title) }
```
Comment (okohuch) [inline spec/models/news_spec.rb]: https://github.com/thoughtbot/shoulda-matchers/blob/master/lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb#L16

File: db/migrate/20180405090900_create_news.rb (hunk 18 above)
Comment (Retttro) [reply db/migrate/20180405090900_create_news.rb]: @okohuch yes

Comment (okohuch) [reply db/migrate/20180405090900_create_news.rb]: and it passes successfully?

Comment (okohuch) [reply db/migrate/20180405090900_create_news.rb]: do you see that something is wrong with this line?

Comment (Retttro) [reply db/migrate/20180405090900_create_news.rb]: @okohuch it passed successfully. maybe delete :news in user_id creating?

File: spec/models/news_spec.rb [hunk 21]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe(News, type: :model) do
+  let(:news) { create(:news) }
+
+  context 'when title field is too short' do
+    it { is_expected.to_not allow_values('No').for(:title) }
```
Comment (okohuch) [inline spec/models/news_spec.rb]: why do you have this test? does not 
```ruby
context 'when title field is minimum length' do
    it { is_expected.to validate_length_of(:title).is_at_least(3) }
end
```
cover it?

Comment (Retttro) [reply spec/models/news_spec.rb]: @okohuch I thought it is needed to test all cases. Will delete that test

Comment (okohuch) [reply spec/models/news_spec.rb]: but it already covered all of them

File: spec/models/news_spec.rb [hunk 22]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe(News, type: :model) do
+  let(:news) { create(:news) }
+
+  context 'when description and title fields are empty' do
+    it do
+      is_expected.to validate_presence_of(:description)
```
Comment (okohuch) [inline spec/models/news_spec.rb]: split into two `it`'s

**Review by okohuch — APPROVED** [review]

**Review by viktoriia-bilous — APPROVED** [review]

**Pull request merged**